### DISCUSSION TITLE
Rechargeable battery situation for ESP32-CYD

### DISCUSSION THREAD

**Discussion by kf106** [witnessmenow/ESP32-Cheap-Yellow-Display#87] [Q&A]
I've been using a Lilygo T-Display-S3 for my hobby work, and you can plug in a LiPo 3.7V battery to the device. On full activity my 300mAh battery lasts about 3 hours. But more importantly, when you plug a USB-C cable into the device, the battery is charged - a red LED lights up, and then goes off when the battery is full.

Is there a battery connector for the ESP32-CYD, and does the USB connector handle its charging properly?

**Comment by witnessmenow** [comment]
No, and as a result no 😅

There is no battery connector or circuitry. A cyd powered by battery will require external modules/components 

I created an issue around this but I haven't made any progress since this 

https://github.com/witnessmenow/ESP32-Cheap-Yellow-Display/issues/47

**Comment by saharco540** [comment]
This could help you out, adjust to 5V
https://www.aliexpress.com/item/32949661746.html

**Comment by BerndHL** [comment]
I just built an esp32-2432s028 with a 3.7V lithium-ion battery, an MT3608 and a TP4056 in a 3D printed case. Unfortunately, so far only between 3 and 5 hours of running time. so far only LVGL display without interactions.

**Reply by AndreLynch** [reply]
That's a pretty good time . I get around 90 minutes

**Reply by slayer522** [reply]
do you have a guide for this?

**Comment by SimpleAsh** [comment]
Hi everyone, I was reading through this A month or so ago, I did a bit of Think and got the bits I thought might work built it this is what I came up with 
![20240724_112744](https://github.com/user-attachments/assets/39256d2a-cedc-46f0-abd3-12e153ec9c9b)
![20240724_113228](https://github.com/user-attachments/assets/eacb5811-f209-4ef3-a529-1480911982e7)
![Messenger_creation_b916ddc5-6f31-4ed9-b005-951fba236cef](https://github.com/user-attachments/assets/068d3397-3826-4e01-9980-45d23e4c5861)
![20240724_164109](https://github.com/user-attachments/assets/3b31ffe3-762d-4225-bb91-f87666ab7b6f)

![20240728_091736](https://github.com/user-attachments/assets/dac938e1-00a2-406b-bd94-87a30aa9f5a2)


Working charge/ discharge same as anything else $1.80 and $7 tidy , thanks for everyone's comments it made this a simple and fast build

![20240728_092142](https://github.com/user-attachments/assets/1f96b9f6-ac6f-48d1-bd6b-cf10f3bfc74b)

![20240728_092152](https://github.com/user-attachments/assets/7f78fd60-ba61-4069-a28c-c339b1885d30)
I

**Reply by SimpleAsh** [reply]
Sorry also fist post here didn't have it planned out

**Reply by SimpleAsh** [reply]
There are a few things I would do different. On the photo with the battery and drawing they are around the wrong way and if you have the same case from AliExpress move the power module all the way to the edge of the case I used hot glue so I might move, and also might cut the case so getting the SD or is a little easier, I also found hot glue the wires to the board in place as they are very small contacts at t back of the USB and I have a very cheap soldering iron,

**Reply by slayer522** [reply]
do you need to have two usb ports? in your video the battery was charging via the pre-installed port. 

Is this because you wanted a dedicated charging port?

**Comment by SimpleAsh** [comment]
![20240728_091736](https://github.com/user-attachments/assets/7e05546c-86a7-4959-a914-e69f1c3555b3)
This is working not sure of time yet only just tried

**Reply by raynoxu1337** [reply]
whats the module you used? Does it boot to 5v? i would like that module for this and some other projects :)

**Comment by SimpleAsh** [comment]
https://youtu.be/XBvBRjAgCMo?si=BMW4hoT7bKckr7gd

**Reply by clach04** [reply]
https://www.youtube.com/watch?v=0xVrZQx4MY4 is worth checking out too.

**Comment by clach04** [comment]
@kf106 is this worth marking as answered now?

**Reply by SimpleAsh** [reply]
Just a bit longer like to see if ups is gd

**Reply by xawos** [reply]
Friendly ping, an year has passed :)
Was this "lil bit longer" enough?

**Comment by SimpleAsh** [comment]
Probably the only thing I can add , and respect for the last YouTube link,


Is never had a problem as it comes with a box , and a reply to others there is a ups board that will fx , plug and play stuff,  ... And anymore new if you're not adding gps ,, listen and read everything, if you guess, you might see the same things

**Reply by SimpleAsh** [reply]
5V1A UPS Uninterruptible Power Supply Module 3.7V Polymer 18650 Lithium Battery Reverse Connection Protection Route Monitoring
https://a.aliexpress.com/_mqsP7W3



Need to remove jumper if not working
